### PR TITLE
add docling and deepsearch-glm

### PR DESCRIPTION
**Pull request by hadim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hadim** [comment]
blocked by:

- [ ] https://github.com/conda-forge/staged-recipes/pull/28092
- [ ] https://github.com/conda-forge/staged-recipes/pull/28091

---

It also requires https://github.com/PeterStaar-IBM/fastText to be packaged (fork without tag of an archived repo).

**Comment by hadim** [comment]
Blocked by https://github.com/conda-forge/python-pptx-feedstock/pull/28

**Comment by hadim** [comment]
@conda-forge/help-python @conda-forge/help-python-c ready for review. Thank you!

The failure on Windows is due to `docling` depending on `deepsearch-glm` (but `deepsearch-glm` is not available on windows).

**Comment by hadim** [comment]
Friendly bump!

File: recipes/deepsearch-glm/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    - rich >=13.7.0
+    - docutils !=0.21
+    - requests
+    - ${{ "pywin32 >=305" if win }}
+
+tests:
+  - python:
+      imports:
+        - deepsearch_glm
+      pip_check: false
```
Comment (h-vetinari) [inline recipes/deepsearch-glm/recipe.yaml:68]: Why is this set to false? If we need to diverge somewhere, we should patch the metadata. If another package is at fault, we should fix it there. The problem is that `pip check` failures are viral and infect all environments that contain any failing package.

File: recipes/docling/recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    - pandas >=2.1.4
+    - marko >=2.1.2
+  run_constraints:
+    - tesserocr >=2.7.1
+
+tests:
+  - python:
+      imports:
+        - docling
+      pip_check: false
```
Comment (h-vetinari) [inline recipes/docling/recipe.yaml:57]: idem

File: recipes/deepsearch-glm/recipe.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    - cxxopts
+    - nlohmann_json
+    - loguru-cpp
+    - utfcpp
+    - fasttext-for-deepsearch-glm
+    - json_schema_validator
+    - pcre2
+    - sentencepiece
+    - pkg-config
+    - zlib
```
Comment (h-vetinari) [inline recipes/deepsearch-glm/recipe.yaml:50]: I'd much prefer some sort of grouping here, e.g.
```
  host:
    # build tools
    - python
    - poetry-core
    - pybind11 >=2.13.1
    - pip
    - pkg-config
    # libraries
    - cxxopts
    - fmt
    - ...
```
and sort the libraries alphabetically. Also, if you're using `sentencepiece` only as a library from C/C++, you probably want to use `libsentencepiece`.

Comment (hadim) [reply recipes/deepsearch-glm/recipe.yaml:50]: good idea. will do.

File: recipes/deepsearch-glm/recipe.yaml (hunk 1 above)
Comment (hadim) [reply recipes/deepsearch-glm/recipe.yaml:68]: will try.

**Comment by hadim** [comment]
@h-vetinari thanks for looking into it.

I will close here and open two new PRs as I am having issues building locally.

**Comment by hadim** [comment]
See

- https://github.com/conda-forge/staged-recipes/pull/28269
- https://github.com/conda-forge/staged-recipes/pull/28270

**Pull request closed without merging**